compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirFunctionParameterChecker.kt

```diff
@@ -70,4 +77,45 @@ object FirFunctionParameterChecker : FirFunctionChecker() {
             }
         }
     }
+
+    private fun checkUninitializedParameterWithEscapedUnderscoreAsName(
```

Comment:
I don't quite understand why this is a special case that isn't covered already with the 1st commit. Can you explain further? Why is escaped underscore special? It's just a name. How is:
```kt
fun foo(x: Int = `_`, `_`: Int)
```
different from:
```kt
fun foo(x: Int = y, y: Int)
```

Reply:
Yeah, the only difference I can tell is, one is resolved, while the other is not, and... that doesn't make sense, right? That's another thing that led me to do a rework: #4321.